app/Console/Commands/VideoData.php

```diff
@@ -39,32 +42,46 @@ class VideoData extends Command
 	 */
 	public function handle(): int
 	{
+		$timeout = intval($this->argument('timeout'));
+		$count = intval($this->argument('count'));
 		try {
-			set_time_limit($this->argument('timeout'));
+			try {
+				set_time_limit($timeout);
+			} catch (InfoException) {
+				// Silently do nothing, if `set_time_limit` is denied.
+			}
 
 			$this->line(
 				sprintf(
 					'Will attempt to generate up to %s video thumbnails/metadata with a timeout of %d seconds...',
-					$this->argument('count'),
-					$this->argument('timeout')
+					$count,
+					$timeout
 				)
 			);
 
-			$photos = Photo::query()
+			$counted = Photo::query()
 				->with(['size_variants'])
 				->whereIn('type', MediaFile::SUPPORTED_VIDEO_MIME_TYPES)
 				->whereDoesntHave('size_variants', function (Builder $query) {
 					$query->where('type', '=', SizeVariant::THUMB);
 				})
-				->take($this->argument('count'))
-				->get();
+				->count();
 
-			if (count($photos) == 0) {
+			if ($counted === 0) {
 				$this->line('No videos require processing');
 
 				return 0;
 			}
 
+			$photos = Photo::query()
+				->with(['size_variants'])
+				->whereIn('type', MediaFile::SUPPORTED_VIDEO_MIME_TYPES)
+				->whereDoesntHave('size_variants', function (Builder $query) {
+					$query->where('type', '=', SizeVariant::THUMB);
+				})
+				->take($count)
+				->get();
+
```

Comment:
Why this duplication of the same query? This only bears the risk that these queries become inconsistent. If the number of results is non-zero, we need to fetch them anyway. If the number of results is zero, the second query is not less efficient than the first.

As far as I see it, the only problem for PhpStan was the `count($photos) == 0)` in the old code. a) PhpStan does not know that `$photos` is countable, b) loos comparision.

The better way would have been to annotate the variable `$photos` with
```php
/** @var Collection<Photo> */
$photos = Photo::query()->
	...
```

Reply:
Humm changed my mind, I was considering that count() should be done in SQL land to avoid going through the collection, but if we are going to use the collection anyway, there is not much point to such optimization.